### ISSUE TITLE
Rule Change: Add suggestions to no-constant-binary-expression

### ISSUE THREAD

**Issue by littlemoji** [eslint/eslint#18051]
### What rule do you want to change?

no-constant-binary-expression

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
function a():boolean{
  ...
}
if(!a){
// !a always return false, and didn't report an error 
  ...
}
```

### What does the rule currently do for this code?

nothing, no error report

### What will the rule do after it's changed?

Report an error

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by Tanujkanti4441** [comment]
Hey @littlemoji, thanks for the issue, i am a bit confuse, your title says 'to add suggestions' but from the code example you have given seems like the issue is about to enable the rule to check the types also, can you clear this.

**Comment by aladdin-add** [comment]
it's confusing: "suggestions" is a term used in eslint: https://eslint.org/docs/latest/extend/custom-rules#providing-suggestions. 😄 

If you're proposing to report an error in the code in the example, this is not possible in eslint because eslint doesn't have type information. But it's possible in ts-eslint. /cc @bradzacher

**Comment by bradzacher** [comment]
You could use scope analysis to attempt to resolve an expression to a static value (I believe there's already a util function you guys use for this purpose!) which would let you provide extra help messages for *some* cases.

**Comment by bradzacher** [comment]
If you want a rule which warns against constant/unnecessary conditions - we have that - https://typescript-eslint.io/rules/no-unnecessary-condition/

**Comment by mdjermanovic** [comment]
> ```js
> function a():boolean{
>   ...
> }
> if(!a){
> // !a always return false, and didn't report an error 
>   ...
> }
> ```

This case is included in https://github.com/eslint/eslint/issues/13776 proposal, so I'm closing this issue as a duplicate.
